core-bundle/src/Controller/BackendPreviewSwitchController.php

```diff
@@ -0,0 +1,149 @@
+<?php
+
+declare(strict_types=1);
+
+/*
+ * This file is part of Contao.
+ *
+ * (c) Leo Feyer
+ *
+ * @license LGPL-3.0-or-later
+ */
+
+namespace Contao\CoreBundle\Controller;
+
+use Contao\BackendUser;
+use Contao\CoreBundle\Exception\PageNotFoundException;
+use Contao\CoreBundle\Framework\ContaoFramework;
+use Contao\CoreBundle\Security\Authentication\FrontendPreviewAuthenticator;
+use Contao\CoreBundle\Security\Authentication\Token\TokenChecker;
+use Contao\Date;
+use Doctrine\DBAL\Connection;
+use Doctrine\DBAL\FetchMode;
+use Symfony\Component\HttpFoundation\JsonResponse;
+use Symfony\Component\HttpFoundation\Request;
+use Symfony\Component\HttpFoundation\Response;
+use Symfony\Component\Routing\Annotation\Route;
+
+/**
+ * This controller serves for the back end preview toolbar by providing two ajax endpoints:
+ * a) Providing members' usernames for the datalist
+ * b) Process the switch action (i.e. log in a specific front end user)
+ *
+ * @Route(defaults={"_scope" = "backend"})
+ */
+final class BackendPreviewSwitchController
+{
+
+    private $contaoFramework;
+
+    private $frontendPreviewAuthenticator;
+
+    private $tokenChecker;
+
+    private $connection;
+
+    public function __construct(
+        ContaoFramework $contaoFramework,
+        FrontendPreviewAuthenticator $frontendPreviewAuthenticator,
+        TokenChecker $tokenChecker,
+        Connection $connection
+    ) {
+        $this->contaoFramework              = $contaoFramework;
+        $this->frontendPreviewAuthenticator = $frontendPreviewAuthenticator;
+        $this->tokenChecker                 = $tokenChecker;
+        $this->connection                   = $connection;
+    }
+
+    /**
+     * @Route("/_contao/preview_switch", name="contao_backend_preview_switch")
```

Comment:
I think the `_contao/` routes are used for the front end only. Maybe we should use `/contao/preview_switch`? That should also make sure the security access control checks for the back end user.